src/rum/config.cpp

```diff
@@ -11,65 +11,72 @@
 namespace {
 
 char *set_config(ngx_conf_t *cf, ngx_command_t *command, void *conf) {
-  auto *rum_config =
-      static_cast<std::unordered_map<std::string, std::string> *>(conf);
+  auto &rum_config =
+      *static_cast<std::unordered_map<std::string, std::vector<std::string>> *>(
+          conf);
 
   if (cf->args->nelts < 2) {
     char *err_msg =
         static_cast<char *>(ngx_palloc(cf->pool, sizeof(char) * 256));
-    ngx_snprintf((u_char *)err_msg, 256,
-                 "invalid number of arguments. Expected exactly two: a key and "
-                 "a value.");
+    ngx_snprintf(
+        (u_char *)err_msg, 256,
+        "invalid number of arguments. Expected at least two arguments.");
     return err_msg;
   }
 
-  ngx_str_t *values = (ngx_str_t *)(cf->args->elts);
+  ngx_str_t *arg_values = (ngx_str_t *)(cf->args->elts);
 
-  std::string_view key = datadog::nginx::to_string_view(values[0]);
+  std::string_view key = datadog::nginx::to_string_view(arg_values[0]);
   if (key.empty()) {
     char *err_msg =
         static_cast<char *>(ngx_palloc(cf->pool, sizeof(char) * 256));
     ngx_snprintf((u_char *)err_msg, 256, "empty key");
     return err_msg;
   }
 
-  std::string_view value = datadog::nginx::to_string_view(values[1]);
-  if (value.empty()) {
-    char *err_msg =
-        static_cast<char *>(ngx_palloc(cf->pool, sizeof(char) * 256));
-    ngx_snprintf((u_char *)err_msg, 256, "empty value");
-    return err_msg;
+  std::vector<std::string> values;
+  for (size_t i = 1; i < cf->args->nelts; ++i) {
+    values.emplace_back(datadog::nginx::to_string(arg_values[i]));
   }
 
-  (*rum_config)[std::string(key)] = std::string(value);
+  rum_config[std::string(key)] = std::move(values);
   return NGX_CONF_OK;
 }
 
 static std::string make_rum_json_config(
     int config_version,
-    const std::unordered_map<std::string, std::string> &config) {
+    const std::unordered_map<std::string, std::vector<std::string>> &config) {
   rapidjson::Document doc;
   doc.SetObject();
 
   rapidjson::Document::AllocatorType &allocator = doc.GetAllocator();
   doc.AddMember("majorVersion", config_version, allocator);
 
   rapidjson::Value rum(rapidjson::kObjectType);
-  for (const auto &[key, value] : config) {
-    if (key == "majorVersion") {
-      continue;
```

Comment:
Curious to know: Is there a reason why we were skipping majorVersion before?

Reply:
one of the first iteration, `majorVersion` was part of the map but I quickly removed it to be a seperate argument. I think it is a left over.